core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -474,68 +421,52 @@ private Collection<DeleteFile> loadDeleteFiles() {
     DeleteFileIndex build() {
       Iterable<DeleteFile> files = deleteFiles != null ? filterDeleteFiles() : loadDeleteFiles();
 
-      boolean useColumnStatsFiltering = false;
+      EqualityDeletes globalDeletes = new EqualityDeletes();
+      PartitionMap<EqualityDeletes> eqDeletesByPartition = PartitionMap.create(specsById);
+      PartitionMap<PositionDeletes> posDeletesByPartition = PartitionMap.create(specsById);
+      CharSequenceMap<PositionDeletes> posDeletesByPath = CharSequenceMap.create();
```

Comment:
I guess its possible we'll hit memory issues with using path map, example if we have a million data files, can be ~100 mbs, if i calculate correctly?.  (We use DeleteFileIndex centrally if i understand correctly).  I wonder if we considered doing some common-prefix encoding, or other encoding, of paths anywhere.  Not sure if its worth it , given the typical memory size of planner

Reply:
It would require a bit of extra space but it would not be possible to plan jobs at all otherwise if we have 1-1 mapping of data and deletes. I am going to add a benchmark. Without this PR, the planning did not complete in 5 minutes. With this PR, it took around 5 seconds.

Reply:
There are also some ideas about storing extra metadata but it will require changing the spec. This can be done later. The current solution should be good enough.